network-store-iidm-impl/src/main/java/com/powsybl/network/store/iidm/impl/ConnectDisconnectUtil.java

```diff
@@ -0,0 +1,182 @@
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.network.store.iidm.impl;
+
+import com.powsybl.commons.report.ReportNode;
+import com.powsybl.commons.report.TypedValue;
+import com.powsybl.iidm.network.Switch;
+import com.powsybl.iidm.network.Terminal;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Predicate;
+
+/**
+ * @author Nicolas Rol {@literal <nicolas.rol at rte-france.com>}
+ */
+public final class ConnectDisconnectUtil {
+
+    private ConnectDisconnectUtil() {
+    }
+
+    /**
+     * Connect the specified terminals. It will connect all the specified terminals or none if at least one cannot be
+     * connected.
+     * @param identifiable network element to connect. It can be a connectable, a tie line or an HVDC line
+     * @param terminals list of the terminals that should be connected. For a connectable, it should be its own
+     *                  terminals, while for a tie line (respectively an HVDC line) it should be the terminals of the
+     *                  underlying dangling lines (respectively converter stations)
+     * @param isTypeSwitchToOperate type of switches that can be operated
+     * @param reportNode report node
+     * @return {@code true} if all the specified terminals have been connected, else {@code false}.
+     */
+    static boolean connectAllTerminals(AbstractIdentifiableImpl<?, ?> identifiable, List<Terminal> terminals, Predicate<Switch> isTypeSwitchToOperate, ReportNode reportNode) {
+
+        // Booleans
+        boolean isAlreadyConnected = true;
+        boolean canBeConnected = true;
+
+        // Initialisation of a list to open in case some terminals are in node-breaker view
+        Set<SwitchImpl> switchForConnection = new HashSet<>();
+
+        // We try to connect each terminal
+        for (Terminal terminal : terminals) {
+            // Check if the terminal is already connected
+            if (terminal.isConnected()) {
+                // If the terminal is already connected, log and continue on other terminals
+                reportNode.newReportNode()
+                    .withMessageTemplate("alreadyConnectedTerminal", "A terminal of identifiable ${identifiable} is already connected.")
+                    .withUntypedValue("identifiable", identifiable.getId())
+                    .withSeverity(TypedValue.WARN_SEVERITY)
+                    .add();
+                continue;
+            } else {
+                isAlreadyConnected = false;
+            }
+
+            // If it's a node-breaker terminal, the switches to connect are added to a set
+            if (terminal instanceof TerminalImpl<?> terminalImpl && terminalImpl.isNodeBeakerTopologyKind()) {
+                canBeConnected = terminalImpl.getConnectingSwitches(isTypeSwitchToOperate, switchForConnection);
```

Comment:
do we really need to do this if terminal.isConnect() returns true ..?

Reply:
if `terminal.isConnected()` returns true, we have a `continue` so this part is skipped

Reply:
Ah yes sorry